src/configuration/yaml.md

```diff
@@ -0,0 +1,169 @@
+
+```yaml
+name: 'app'
+type: 'php:7.1'
+build:
+    flavor: 'composer'
+
+disk: 1024
+```
+
+This example defines a key `name` with value `app`, a key `type` with value `php:7.1`, a key `disk` with a value 1024, and a key `build` that is itself a nested set of key/value pairs, of which there is only one: `flavor`, whose value is `composer`.  Nested values are often referenced using a dotted syntax, such as `build.flavor`.
+
+Keys are always strings, and may be quoted or not.  Values may be strings, numbers, or booleans.  Alphanumeric strings may be quoted or not.  More complex strings (with punctuation, etc.) must be quoted.  Numbers should not be quoted.  The boolean values `true` and `false` should never be quoted.
+
+For quoted values, both single quotes (`'`) and double quotes (`"`) are valid.  Double quotes, however, will interpolate common escape characters such as `\n` and so forth.  For that reason using single quotes is generally recommended unless you want escape characters to be processed rather than taken literally.
+
+In general the order of keys in a YAML file does not matter.  Neither do blank lines.  Indentation may be with any number of spaces, as long as it is consistent throughout the file.  Platform.sh examples by convention use four-space indentation.
+
+## Multi-line strings
+
+In case of long, multi-line strings, the `|` character tells the YAML parser that the following, indented lines are all part of the same string.  That is, this:
+
+```yaml
+hooks:
+    build: |
+        set -e
+        cp a.txt b.txt
+```
+
+creates a nested property `hooks.build`, which has the value `set -e\ncp a.txt b.txt`.  (That is, a string with a line break in it.)  That is useful primarily for hooks, which allow the user to enter small shell scripts within the YAML file.
+
+## Includes
+
+YAML allows for special "tags" on values that change their meaning.  These tags may be customized for individual applications so may vary from one system to another.  The main Platform.sh "local tag" is `!include`, which allows for external files to be logically embedded within the YAML file.  The referenced file is always relative to the YAML file's directory.
+
+### `string`
+
+The `string` type allows an external file to be inlined in the YAML file as though it had been entered as a multi-line string.  For example, given this file on disk named `build.sh`:
+
+```text
+set -e
+cp a.txt b.txt
+```
+
+Then the following two YAML fragments are exactly equivalent:
+
+```yaml
+hooks:
+    build: |
+        set -e
+        cp a.txt b.txt
+```
+
+```yaml
+hooks:
+    build: !include
+      type: string
+      path: build.sh
+```
+
+That is primarily useful for breaking longer build scripts or inlined configuration files out to a separate file for easier maintenance.
```

Comment:
But this would be simpler, if build.sh is executable:

```yaml
hooks:
    build: ./build.sh
```

Reply:
... True, that would be even better.  I don't want to get into the Varnish example here, though, especially as that's not published yet.  Have you a better suggestion for an example to use?